stave/views.py

```diff
@@ -58,6 +58,62 @@ def get_queryset(self) -> QuerySet[models.Application]:
         return super().get_queryset().filter(user=self.request.user)
 
 
+class OfficiatingHistoryView(LoginRequiredMixin, generic.ListView):
+    """
+    A view that displays a user's officiating history.
+
+    The QuerySet fetches all RoleGroupCrewAssignments with a crew that has the user in a crew
+    assignment. `get_context_data` then computes the effective crew for each of those and bulids a
+    list of GameHistory's. This requires only a single query, and a bit of filtering in memory.
+
+    The QuerySet can be paginated, although the number of game histories shown on the results could
+    be less than the size of the page because the effective crew may not include the user, and could
+    even theoretically be empty. This could be fixed by having get_queryset return GameHistory's,
+    but it's not clear if that can be composed as a QuerySet.
+    """
+
+    template_name = "stave/officiating_history.html"
+    model = models.RoleGroupCrewAssignment
+
+    def get_queryset(self):
+        cas = models.CrewAssignment.objects.filter(user=self.request.user)
+        crews = models.Crew.objects.filter(assignments__in=cas)
+        rgcas = models.RoleGroupCrewAssignment.objects.filter(
+            Q(crew__in=crews) | Q(crew_overrides__in=crews)
+        ).order_by("-game__start_time")
+        return rgcas
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+
+        histories = []
+        for rgca in context["object_list"]:
+            cas = [ca for ca in rgca.effective_crew() if ca.user == self.request.user]
```

Comment:
Yeah, this is a nice way to handle the "user might be on a static crew with an override" problem. :+1: